### PR TITLE
PXD-1548 - Chore/revproxy

### PR DESCRIPTION
**Pull request by frickjack**
* decompose the reverse proxy nginx config into multiple files - one per service
* patch `kube-setup-revproxy` to only include the active services in the `revproxy-subconf` configmap
* putch `gen3 roll all` to skip deploying services that do not have an entry in the cdis-manifest `versions` block
* change the default `auth field` in the arranger deployment
* add CORS headers to revproxy
* add fence sidecar to arborist

### CODE REVIEW

File: gen3/bin/es.sh [hunk 1]
```diff
@@ -147,15 +147,44 @@ function es_import() {
...
+    npx elasticdump --output http://$ESHOST/$name --input $sourceFolder/${name}__data.json --type data
     let importCount+=1
   done
   if [[ $importCount == 0 ]]; then
-    echo "ERROR: no .json files found matching $projectName" 1>2
+    echo "ERROR: no .json files found matching $projectName" 1>&2
     return 1
   fi
   # make sure arranger-projects index has an entry for our project id
+  if ! gen3 es indices | awk '{ print $3 }' | grep -e '^arranger-frickjacks$' > /dev/null 2>&1; then
```
Comment (philloooo) [inline gen3/bin/es.sh]: why your name is here :p

Comment (frickjack) [reply gen3/bin/es.sh]: doh!  fixed it ...

File: gen3/bin/kube-setup-revproxy.sh [hunk 2]
```diff
@@ -32,4 +47,49 @@ else
   echo "Ensure the commons DNS references the -elb revproxy which support http proxy protocol"
 fi
 
-bash "${GEN3_HOME}/kube/services/revproxy/apply_service"
+DRY_RUN=${DRY_RUN:-""}
```
Comment (philloooo) [inline gen3/bin/kube-setup-revproxy.sh]: can you add explaination in revproxy/README.md about what the `-dry-run`is for?

File: kube/services/revproxy/gen3.nginx.conf/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# TL;DR
+
+Decomposed nginx reverse proxy configuration.  Use `gen3 kube-setup-revproxy` to update the revproxy config.
```
Comment (philloooo) [inline kube/services/revproxy/gen3.nginx.conf/README.md]: can you add some brief explaination about how it works, eg: it's based on checking whether the k8s service is up for each api

Comment (frickjack) [reply kube/services/revproxy/gen3.nginx.conf/README.md]: Good idea.  Done!

**Pull request merged**